kahuna/public/js/upload/controller.js

```diff
@@ -46,7 +46,7 @@ upload.controller('UploadCtrl', [
 
         ctrl.onDeleteError = function (err) {
             if (err.body.errorKey === 'image-not-found') {
-                $state.go('search');
+                $state.go('upload', {}, {reload: true});
```

Comment:
> Is that going to cause going from Your current uploads to Last 50 uploads?

This is within the `Last 50 Uploads`... if a delete fails, reload the page rather than go to search page, which is better behaviour imo.